i3pystatus/pulseaudio/__init__.py

```diff
@@ -198,12 +198,9 @@ def sink_info_cb(self, context, sink_info_p, eol, _):
                 raise Exception("bar_type must be 'vertical' or 'horizontal'")
 
             selected = ""
-            dump = subprocess.check_output("pacmd dump".split(), universal_newlines=True)
-            for line in dump.split("\n"):
-                if line.startswith("set-default-sink"):
-                    default_sink = line.split()[1]
-                    if default_sink == self.current_sink:
-                        selected = self.format_selected
+            default_sink = subprocess.check_output("pactl get-default-sink".split()).strip()
```

Comment:
If pulseaudio is for some reason not running, and `pactl` is run, will it return a nonzero exit code? Because if so, `subprocess.check_output()` will raise an exception that you are not catching here.

Reply:
The same issue exists with pacmd and the original code doesn't catch the exception either. I ran with both this version and the version with pacmd and got the same output when pulse wasn't running. It appears to just run as if the pulseaudio module isn't registered.

Reply:
Hi,

Apologies for the delay in responding to this PR. I received an email notification a day ago regarding a comment from a GitHub user who pointed out that the current code is already causing an error if `pactl` is not present. However, it appears that the user has since deleted their comment.

In my view, hiding this error might not be beneficial for end users, as it could indicate a configuration issue with `pactl`. I propose three options:
1. Keep the code as is, assuming users will investigate their system configuration.
2. Display the error differently.
3. Hide the error entirely, though I believe this is not ideal.

I'd like this PR to be merged. @terminalmage, what's your preference among these options? Or maybe you would like to something else entirely?

Kind regards,

Reply:
My preference is to write the error to the logger. I am a big fan of using a logfile in the i3pystatus script. I think the [configuration docs](https://i3pystatus.readthedocs.io/en/latest/configuration.html) should probably be updated to add a `logfile` argument when instantiation your `Status` instance, but that's another discussion entirely (maybe I'll open a PR for that :smile:).

Each `IntervalModule` subclass instance already has a logger assigned to an attribute, so you could use something like:

```python
self.logger.error("some error")
```

If you are logging from within an `except` block, then you can use `self.logger.exception("some error")`, which is equivalent to `self.logger.error("some message", exc_info=True)`. This adds the original traceback to the log file, and logs at the `ERROR` loglevel.